### PR TITLE
add error middleware

### PR DESCRIPTION
**Pull request by maxired**
Hi,

this is my attempt to fix https://github.com/amaurymartiny/now-middleware/issues/25

I don't have any TS background, so sorry if missing something ;-)
This seems to work great.

### CODE REVIEW

File: src/chain.spec.ts [hunk 1]
```diff
@@ -46,4 +46,86 @@ describe('chain', () => {
...
+
+    function errorCatchMiddleware(
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      err: any,
+      _req: Request,
+      _res: Response,
+      next: NextFunction
+    ): void {
+      if (err) {
+        _res.send(err.message);
```
Comment (amaurymartiny) [inline src/chain.spec.ts:68]:
```suggestion
        res.send(err.message);
```

File: src/chain.spec.ts [hunk 2]
```diff
@@ -46,4 +46,86 @@ describe('chain', () => {
...
```
Comment (amaurymartiny) [inline src/chain.spec.ts:109]:
```suggestion
        res.send(err.message);
```

File: src/chain.ts [hunk 3]
```diff
@@ -24,26 +30,48 @@ type AsyncVoid = void | Promise<void>;
...
+      if (err) {
+        if (mid.length === 4) {
+          // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+          // @ts-ignore
+          return mid(err, req, res, next);
+        }
+        return next(err);
+      }
+      if (mid.length === 4) {
+        return next();
```
Comment (amaurymartiny) [inline src/chain.ts:48]: Shouldn't it be?

```ts
    if (mid.length === 4) {
      try {
        // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
        // @ts-ignore
        mid(undefined, req, res, next);
      } catch (e) {
        next(e);
      }
    }
```

i.e. any particular reason to ignore middlewares with 4 arguments?

Comment (maxired) [reply src/chain.ts:48]: Middlewares with 4 arguments are error handlers, and I believe they should not be run if no errors

Comment (maxired) [reply src/chain.ts:48]: this is my understanding from https://github.com/expressjs/express/blob/3ed5090ca91f6a387e66370d57ead94d886275e1/lib/router/layer.js#L89-L92

File: src/chain.spec.ts (hunk 2 above)
Comment (maxired) [reply src/chain.spec.ts]: fixed in https://github.com/amaurymartiny/now-middleware/pull/26/commits/4f78b37547ebeb49d2db2079fc6db2192949cc60

File: src/chain.spec.ts (hunk 1 above)
Comment (maxired) [reply src/chain.spec.ts]: fixed in https://github.com/amaurymartiny/now-middleware/pull/26/commits/4f78b37547ebeb49d2db2079fc6db2192949cc60

File: src/chain.ts (hunk 3 above)
Comment (amaurymartiny) [reply src/chain.ts:48]: Ah nice, makes sense 👍

**Comment by maxired** [comment]
thanks a lot for the fast review and release!

**Pull request merged**